examples/components/CCFRAUD_VERTICAL/traininsilo/aml_comm.py

```diff
@@ -316,32 +304,49 @@ def send(self, data, destination):
             data: data to be sent
             destination: rank of the receiver node
         """
+        assert destination != self._rank
```

Comment:
can you just add some error message here to explain why that's happening? (it won't, but just in case)